internal/server/kong/ws/config/compat/compatibility_table.go

```diff
@@ -634,8 +635,9 @@ var (
 						Condition: "functions",
 						Updates: []config.ConfigTableFieldUpdate{
 							{
-								Field:          "access",
-								ValueFromField: "functions",
+								Field:            "access",
+								ValueFromField:   "functions",
+								FieldMustBeEmpty: true,
```

Comment:
Since we are switching from `break` to `continue` to ensure the ordering doesn't mess things up between removal and updates we should re-add:

```
DisableChangeTracking: func(rawJSON string) bool {
	// do not emit change if functions is set to default value (empty array)
	plugin := gjson.Parse(rawJSON)
	return len(plugin.Get("config.plugin_field_1").Array()) == 0
},
```

Reply:
Switching `break` to `continue` here also requires switching from `break` to `continue` [here](https://github.com/Kong/koko/blob/main/internal/server/kong/ws/config/version_compatibility.go#L610), which causes [this test](https://github.com/Kong/koko/blob/main/internal/server/kong/ws/config/version_compatibility_test.go#L5409) to fail. Not switching means [ his test](https://github.com/Kong/koko/blob/18e185dd9efd41e76298224671f9716a486c925b/internal/server/kong/ws/config/version_compatibility_test.go#L5845-L5902) case will fail:
```
{
			name: "field updates do not create new fields when ignoring empty array values",
			configTableUpdates: map[string][]ConfigTableUpdates{
				">= 3.0.0": {
					{
						Name: "plugin_1",
						Type: Plugin,
						DisableChangeTracking: func(rawJSON string) bool {
							plugin := gjson.Parse(rawJSON)
							return ValueIsEmpty(plugin.Get("config.plugin_field_1"))
						},
						FieldUpdates: []ConfigTableFieldCondition{
							{
								Field:     "plugin_field_1",
								Condition: "plugin_field_1",
								Updates: []ConfigTableFieldUpdate{
									{
										Field:            "plugin_field_2",
										ValueFromField:   "plugin_field_1",
										FieldMustBeEmpty: true,
									},
									{
										Field: "plugin_field_1",
									},
								},
							},
						},
						ChangeID: "T101",
					},
				},
			},
			uncompressedPayload: `{
				"config_table": {
					"plugins": [
						{
							"id": "759c0d3a-bc3d-4ccc-8d4d-f92de95c1f1a",
							"name": "plugin_1",
							"config": {
								"plugin_field_1": []
							}
						}
					]
				}
			}`,
			dataPlaneVersion: "3.0.0",
			expectedPayload: `{
				"config_table": {
					"plugins": [
						{
							"id": "759c0d3a-bc3d-4ccc-8d4d-f92de95c1f1a",
							"name": "plugin_1",
							"config": {}
						}
					]
				}
			}`,
			expectedChanges: TrackedChanges{},
		},
```
The second, implicit field removal is not processed. I'm looking for a workaround.

Reply:
Actually, I just checked the above test case against main and it fails. Does the implicit field removal actually work?
```
Error:      	Not equal: 
        	            	expected: map[string]interface {}{"config_table":map[string]interface {}{"plugins":[]interface {}{map[string]interface {}{"config":map[string]interface {}{}, "id":"759c0d3a-bc3d-4ccc-8d4d-f92de95c1f1a", "name":"plugin_1"}}}}
        	            	actual  : map[string]interface {}{"config_table":map[string]interface {}{"plugins":[]interface {}{map[string]interface {}{"config":map[string]interface {}{"plugin_field_2":[]interface {}{}}, "id":"759c0d3a-bc3d-4ccc-8d4d-f92de95c1f1a", "name":"plugin_1"}}}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -4,3 +4,5 @@
        	            	    (map[string]interface {}) (len=3) {
        	            	-    (string) (len=6) "config": (map[string]interface {}) {
        	            	+    (string) (len=6) "config": (map[string]interface {}) (len=1) {
        	            	+     (string) (len=14) "plugin_field_2": ([]interface {}) {
        	            	+     }
        	            	     },
```

Reply:
I'm removing the two failing tests in [ecff883](https://github.com/Kong/koko/pull/469/commits/ecff883f22298c4a60cc2c5fc5e636848d2d7e47).

Reply:
This test fails in main too because you need to at least set one field for the plugin:

```
{
    "code": 3,
    "details": [
        {
            "@type": "type.googleapis.com/kong.admin.model.v1.ErrorDetail",
            "messages": [
                "at least one of these fields must be non-empty: 'config.functions', 'config.certificate', 'config.rewrite', 'config.access', 'config.header_filter', 'config.body_filter', 'config.log'"
            ],
            "type": "ERROR_TYPE_ENTITY"
        }
    ],
    "message": "validation error"
}
```

All good then.